exercises/practice/rational-numbers/rational-numbers.test.ts

```diff
@@ -1,197 +1,288 @@
 import { Rational } from './rational-numbers'
 
+function assertSameRational(r1: Rational, r2: Rational): void {
+  expect(r1.numerator).toEqual(r2.numerator)
+  expect(r1.denominator).toEqual(r2.denominator)
+}
+
+describe ('Constructor', () => {
+  it('Assigns public fields numerator and denominator', () => {
+    const rational = new Rational(5, 7)
+    expect(rational.numerator).toEqual(5)
+    expect(rational.denominator).toEqual(7)
+  })
+})
```

Comment:
I had to add this extra test because, even requiring the fields to be public, leaving the fields empty or always equal to 0/1 would allow somebody to make all comparisons pass like the ticket https://github.com/exercism/typescript/issues/1049 mentions

Reply:
Yes, correct. Sorry should have listed that.

You can still fake it now by setting it to 5 and 7, always. Perhaps `assertSameRational` is changed to:

```typescript
assertSameRational(actual, 7, 6)
```

So at each stage we check the actual values?

If you want, I can write to code to be:

```typescript
expect(actual).toBeRational(7, 6)
```

Reply:
Is this better https://github.com/exercism/typescript/pull/1277/commits/b6d4ebc1d2927862db897cbe38758c38b520e3f8 ?

Reply:
I think so, yeah! Hope you're also good with this.